Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -31,6 +31,360 @@
+        status (JobStatus): The job's status.
+        id (str): The job's id in Splunk server.
+        creation_time (str): The job's creation time in ISO format.
+
+    """
+    def __init__(self, job_type, status=None, job_id=None, creation_time=None):
+        self.type = job_type
+        self.id = job_id
+        self.creation_time = creation_time
+        self.status = status if status else JobStatus.IN_PROGRESS
+
+    @property
+    def id(self):
+        return self.id
+
+    @id.setter
+    def id(self, job_id):
+        self.id = job_id
+
+    @property
+    def creation_time(self):
+        return self.creation_time
+
+    @creation_time.setter
+    def creation_time(self, creation_time):
+        self.creation_time = creation_time
+
+    @property
+    def status(self):
+        return self.status
+
+    @status.setter
+    def status(self, status):
+        self.status = status
+
+    def init_creation_time(self):
+        self.creation_time = datetime.utcnow().isoformat()
+
+    @classmethod
+    def from_json(cls, job_dict):
+        """ Deserialization method.
+
+        Args:
+            job_dict: The job dict in JSON format.
+
+        Returns:
+            An instance of the Job class constructed from JSON representation.
+
+        """
+        return cls(
+            job_type=job_dict.get(TYPE),
+            status=job_dict.get(STATUS),
+            job_id=job_dict.get(ID),
+            creation_time=job_dict.get(CREATION_TIME)
+        )
+
+
+class NotableStatus(object):
+    """ Contains all notable's possible status during the enrichment process """
+    ENRICHED = 'Notable has enriched'
+    NOT_YET_ENRICHED = 'Notable has not yet enriched'
+    CREATED_INCIDENT_ENRICH_FAILURE = 'Incident was created from the raw notable'
+
+
+class Notable:
+    """ A class to represent a Notable.
+
+    Args:
+        data (dict): The notable's data.
+        notable_id (str): The notable's ID.
+        status (str): The notable's status.
+
+    Attributes:
+        data (dict): The notable's data.
+        id (str): The notable's id in Splunk server.
+        status (NotableStatus): The notable's status.
+
+    """
+    def __init__(self, data, notable_id=None, status=None):
+        self.data = data
+        self.status = status if status else NotableStatus.NOT_YET_ENRICHED
+        if notable_id:
+            self.id = notable_id
+        else:
+            if EVENT_ID not in data:
+                raise Exception('When using the enrichment mechanism, an event_id field is needed, and thus, '
+                                'one must use a fetch query of the following format: search `notable` .......\n'
+                                'Please re-edit the fetchQuery parameter in the integration configuration, reset '
+                                'the fetch mechanism using the splunk-reset-enriching-fetch-mechanism command and '
+                                'run the fetch again.')
+            self.id = data[EVENT_ID]
+
```

Comment:
Need to add here a function called: `get_notable_job_status` and let's talk about it before you start implementing it.